FEAT: inning endpoint (resolves issues in #88)

reddigari opened the pull request:
This is a resolution to #88, and includes @frankmowen's commits.

As @dgvigil figured out, most of the test failures arose from asserting the type of various attributes. In some cases, the API provides fields that are typically floats as integers. The logic in `mlbgame.object.setobjattr()` casts these to ints successfully, so the assertion that it's a float fails. I figure if the API provides both, we can just check that they're either an int or a float and move on.

In another case, a field that contains an underscore (e.g. `"20180802_191259"`) also has its values succesfully cast to ints (`20180802191259`). I'm less comfortable with this one, as I assume the original string might be able to be used as a identifier for lookups elsewhere. But for now I just changed to the test to expect the int.

I really like the dict-to-object automation of `mlbgame.object.Object`, but we might consider passing a dict to its constructor that specifies the type of certain fields. That would let us selectively control casts like the underscore example above.

On an unrelated note, @panzarino are you cool having the `Inning`, `AtBat`, and `Pitch` objects being subclassed in the `mlbgame.innings` module? I got real confused at first, but then realized that the innings endpoint being used here is basically a more detailed view of the events endpoint. Still not sure there's a benefit to subclassing in `innings.py` since nothing is changing except the pretty-print strings, but if you decide to leave it, I'd recommend adding a subclass of `Action` too. (I think #88 was opened before `Action` was implemented. The tests here don't yet test `Action` objects, which should probably be addressed as well.)

reddigari commented:
So it looks like because `innings.Inning` subclasses `events.Inning` without overriding `__init__`, the object gets constructed with the `events.AtBat` and `events.Pitch` classes rather than their counterparts in `innings.py`. That's why the `__str__` test was failing on the Pitch object. I'm gonna remove that test for the time being.

I definitely think we should avoid the subclasses, and just put logic in the `__str__` method to check for the more detailed fields and use them if available. Gonna wait for @panzarino to weight in before I fidget with it more.

panzarino commented:
I would suggest using the same objects for both (and modifying the `__str__`) if necessary. However, this might be confusing for end users since two objects of the same type may not have the same properties.

reddigari commented:
From what I can tell the innings endpoint has exactly the same information as the events endpoint, but with more detail on pitches, and also `<runner>` tags that capture any runner movements. Would you be okay replacing the `game_events()` data source with the innings endpoint and removing `game_innings()`?

panzarino commented:
If it contains the same data (as to not break existing implementations) and more, then I would rather just replace the events with innings data (and leave it as game_events so that no implementations are broken).

reddigari commented:
There were a couple of attributes missing from the innings endpoint, so I added a parameter to `game_events()` to specify the endpoint. Defaults to `innings_endpoint=False`, which behaves as before. Using `True` replicates the behavior of `game_innings()`, which I removed. The docstring explains that setting the flag to True can result in both additional and missing attributes.

The `AtBat`, `Action`, and `Pitch` classes now have an `_endpoint` attribute that tracks which endpoint provided the data (`'game_events',` or `'innings'`).

I think this is good to go if you approve.

reddigari commented on mlbgame/events.py:
```diff
@@ -200,6 +205,43 @@ class Pitch(mlbgame.object.Object):
         sv_id
         type
+
+    Additional properties if `self._endpoint == 'innings'`:
```
This might break your doc generator.

panzarino commented:
Thanks so much! I'm planning on making a new release sometime this week so these changes will be pushed into production when that is done.
